### PR TITLE
[RUN-5628] Route WebContents events over to BrowserView

### PR DESCRIPTION
**Pull request by MichaelMCoates**
#### Description of Change

This PR Adds the following events from WebContents over to BrowserView:
- page-title-updated (also added to Window)
- did-change-theme-color (also added to Window)
- page-favicon-updated (also added to Window)
- crashed

Tests:
https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5d8a81a67ba87401eaf566a8

Corresponding js-adapter PR:
https://github.com/HadoukenIO/js-adapter/pull/350

#### Checklist

- [X] PR description included and stakeholders cc'd
- [X] `npm test` passes
- [X] automated tests are [changed or added](https://testing-dashboard.openfin.co/#/app/dashboard)
- [X] relevant documentation is [changed or added](https://github.com/hadoukenio/js-adapter)
- [X] PR title starts with the JIRA ticket [pull request process](https://github.com/openfin/Internal-Wiki/wiki/Pull-Request-Process)
- [X] PR release notes describe the change in a way relevant to app-developers
- [X] Link to new tests added
- [X] PR has assigned reviewers

#### Release Notes

Notes: 
N/A, experimental

### CODE REVIEW

**Comment by MichaelMCoates** [comment]
Fixed merge conflict.

**Comment by pbaize** [comment]
Is there a reason we can't share those listeners between window and view?

**Comment by MichaelMCoates** [comment]
> Is there a reason we can't share those listeners between window and view?

window.js has some custom behavior that browserview doesn't have right now. Now that event propagation is in, now would be a good time to discuss how we want to handle these.
https://github.com/HadoukenIO/core/blob/develop/src/browser/api/window.js#L548

**Comment by pbaize** [comment]
could we have webcontents emit to of_events and window.js listens to either webcontents or of_events for the side-effects but doesn't re-emit the event?

**Comment by MichaelMCoates** [comment]
> could we have webcontents emit to of_events and window.js listens to either webcontents or of_events for the side-effects but doesn't re-emit the event?

Done! Waiting for testing.

File: src/browser/api/webcontents.ts [hunk 1]
```diff
@@ -19,11 +19,7 @@ export function hookWebContentsEvents(webContents: Electron.WebContents, { uuid,
     ) => {
         const type = 'resource-response-received';
 
-        const payload = {
-            name,
-            uuid,
-            topic,
-            type,
+        const payload = { uuid, name, topic, type,
```
Comment (rdepena) [inline src/browser/api/webcontents.ts:22]: :neckbeard: why single line ?

File: src/browser/api/webcontents.ts [hunk 2]
```diff
@@ -35,25 +31,70 @@ export function hookWebContentsEvents(webContents: Electron.WebContents, { uuid,
...
         validatedURL,
         isMainFrame
     ) => {
         const type = 'resource-load-failed';
-        const payload = {
-            name,
-            uuid,
-            topic,
-            type,
+        const payload = { uuid, name, topic, type,
```
Comment (rdepena) [inline src/browser/api/webcontents.ts:42]: :neckbeard:

File: src/browser/api/webcontents.ts (hunk 1 above)
Comment (MichaelMCoates) [reply src/browser/api/webcontents.ts:22]: Lol it's the only way to pass the linter. 100 line limit.

**Pull request merged**